Libs/vtkSegmentationCore/vtkSegmentation.cxx

```diff
@@ -308,6 +358,30 @@ std::string vtkSegmentation::GenerateUniqueSegmentID(std::string id)
   return this->GenerateUniqueSegmentID(id + "_");
 }
 
+//---------------------------------------------------------------------------
+std::string vtkSegmentation::GenerateRandomSegmentID(int suffix_Length, std::string validCharacters/*=""*/)
+{
+  if (suffix_Length <= 0)
+  {
+    vtkErrorWithObjectMacro(nullptr, "GenerateRandomSegmentID: Invalid suffix length, must be greater than 0");
+    return "";
+  }
+
+  if (validCharacters.empty())
+  {
+    validCharacters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  }
+
+  vtkMinimalStandardRandomSequence* randomSequence = vtkSegmentation::GetSegmentIDRandomSequenceInstance();
+  std::string randomString = "S_";
+  for (int i = 0; i < suffix_Length; ++i)
+  {
+    int index = static_cast<int>(std::floor(randomSequence->GetNextValue() * validCharacters.size()));
```

Comment:
In theory, we may get the `9` character, but practically it would never happen, as we would just get very close to 1.0 but not actually get 1.0. It would be nice if we would get any of the valid characters with about the same probability.

Reply:
Actually we get 9 as often as the other numbers, however I think that there is a low probability that we will go out of bounds if we get 1.0. Either way I'll update this.